### PR TITLE
Explain analyz

### PR DESCRIPTION
**Pull request by Iqbal-Elham**
What does this PR do?
In this PR I create a Database ...

Description of the task to be completed 👎

In this project we will have a chance to optimize some slow queries in our database.

Feel free to check the query result screenshot [Screenshot Link] (https://github.com/mrkamin/Vet-Clinic/tree/explain-analyz/quiry-result/query-explain-analyz).

Screen Shot of query Results.

query before Apply Index for explain analyze SELECT COUNT(*) FROM visits where animal_id = 4;
explain-analayz-1

query after Apply Index for explain analyze SELECT COUNT(*) FROM visits where animal_id = 4;
explain-analayz-6

query before Apply Index for explain analyze SELECT * FROM visits where vet_id = 2;
explain-analayz-4

query after Apply Index for explain analyze SELECT * FROM visits where vet_id = 2;
explain-analayz-9

query before Apply Index for explain analyze SELECT * FROM owners where email = '[owner_18327@mail.com](mailto:owner_18327@mail.com)';
explain-analayz-5

query after Apply Index for explain analyze SELECT * FROM owners where email = '[owner_18327@mail.com](mailto:owner_18327@mail.com)';
explain-analayz-7

- In this PR Mr. Mohammad Rafi has more commits which merged in dev branch right away.

### CODE REVIEW

**Review by Olamarx — CHANGES_REQUESTED** [review]
Hi Team, 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project, that does not have the commit of all the members of this team, kindly make your third collaborator make some commits too.

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

@Olamarx 
thanks for your good and valuable feedback.

**Review by tugsem — CHANGES_REQUESTED** [review]
##  STATUS: REQUIRED CHANGES ♻️
Hello @Iqbal-Elham, @miltonHenschel and @mrkamin 👋 , 

Good job so far! 👏 
There are some issues that you still need to work on to go to the next project but you are almost there!

Alt Text
## Highlights 🚀 
✔️correct use of gitflow
✔️professional documentation
## Required Changes ♻️

- [x] Kindly add your solutions to the schema.sql file.

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
@tugsem,
thanks for your good and valuable feedback. the solution add to the explain-analyz-Schemas.sql file.
reviewr feedback

File: Schemas.sql [hunk 1]
```diff
@@ -73,4 +73,17 @@ CREATE TABLE visits (
...
+CREATE TABLE visits(
+  id INT GENERATED ALWAYS AS IDENTITY,
+  animal_id INT REFERENCES animals(id),
+  vet_id INT REFERENCES vets(id),
+  date_of_visit DATE,
+  PRIMARY KEY(id)
+);
+
+-- Add an email column to your owners table
+ALTER TABLE owners ADD COLUMN email VARCHAR(120);
```
Comment (danmainah) [inline Schemas.sql:89]: - [x] @team great work trying to implement the changes requested, however, I can't see the code for the solution you implemented to reduce the execution time kindly add it as previously requested

**Review by danmainah — CHANGES_REQUESTED** [review]
Hi @Team, 

Great work on making the changes requested by a previous reviewer 👏  👏 👏
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Highlights

✔️ Gitflow was correctly used
✔️  Good design


 # Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: Schemas.sql (hunk 1 above)
Comment (mrkamin) [reply Schemas.sql:89]: @danmainah,
thanks for your good and valuable feedback. 
we take your comment into account and add the solution to the explain-analyz-Schemas.sql file.
reviewr feedback

**Review by rotimiazeez — APPROVED** [review]
Hi TEAM 👏🏽, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlight 👍🏽
- Good job implementing the suggested changes by the previous CRs

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend considering them as they can improve your code. The previous reviewer simply missed some of them, and addressing them will improve your application._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something needs to be 100% clear.
Remember to tag me(@rotimiazeez) in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561), you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
@rotimiazeez,
thanks for your good and valuable feedback

**Pull request merged**